wasp/widgets.py

```diff
@@ -285,3 +285,80 @@ def touch(self, event):
         elif v >= self._steps:
             v = self._steps - 1
         self.value = v
+
+
+_message_string_x_coord = const(0)
+_message_string_y_coord = const(60)
+_yes_button_x_coord = const(20)
+_yes_button_y_coord = const(100)
+_no_button_x_coord = const(120)
+_no_button_y_coord = const(100)
+
+class ConfirmationView:
+    "Confirmation widget allowing user confirmation of a setting"
+
+    class TouchResult:
+        YES = 0
+        NO = 1
+        NO_ACTION = 2
+
+    def __init__(self):
+        self.drawable = wasp.watch.drawable
+
+        self.yes_button_bounds = (
+            (_yes_button_x_coord, _yes_button_y_coord),
+            (
+                icons.yes_button[0] + _yes_button_x_coord,
+                icons.yes_button[1] + _yes_button_y_coord,
+            ),
+        )
+        self.no_button_bounds = (
+            (_no_button_x_coord, _no_button_y_coord),
+            (
+                icons.no_button[0] + _no_button_x_coord,
+                icons.no_button[1] + _no_button_y_coord,
+            )
+        )
+
+    def draw(self, message):
+        self.drawable.fill(1)
+        self.drawable.string(
+            message,
+            _message_string_x_coord,
+            _message_string_y_coord
+        )
+        self.drawable.blit(
+            icons.yes_button,
+            _yes_button_x_coord,
+            _yes_button_y_coord,
+        )
+        self.drawable.blit(
+            icons.no_button,
+            _no_button_x_coord,
+            _no_button_y_coord,
+        )
+
+
+    def touch(self, event):
+        x_coord = event[1]
+        y_coord = event[2]
+        is_yes_button_press = (
+            x_coord > self.yes_button_bounds[0][0]
+            and y_coord > self.yes_button_bounds[0][1]
+            and x_coord < self.yes_button_bounds[1][0]
+            and y_coord < self.yes_button_bounds[1][1]
+        )
```

Comment:
This seems a little over precise (making the user hit a precise hit box doesn't really help us figure out their intent... it just means that have to press again). For this type of dialog it might be better to divide the screen info quarters and position the button to bottom left implies yes and bottom right implies no.

Reply:
To me this makes sense as the intent is conveyed via the message we sent to the view, the value is what's captured from the resulting `touch()` method. I think I'd find it weird as a user if I touch a out of bounds area of the screen and it still selected yes/no on the result. I know accidentally tap parts of my screen all the time.

Reply:
It's not really a matter of accepting out-of-bounds touches. More about designing the UI to afford a larger hit box. As a simple example consider the launcher app which is designed without hard borders so that accepting presses in simple quardrents matches users expectation.